### PR TITLE
check if lintr is new enough

### PR DESCRIPTION
**Pull request by randy3k**
> I just realized that there is no roadmap for the release of new version of lintr. Perhaps we should include your original trick of checking the text parameter for now? Otherwise we won't know how much time they need to make a new release.

related: #284 

I have decided to use `packageVersion` at the end.

### CODE REVIEW

File: R/diagnostics.R [hunk 1]
```diff
@@ -5,6 +5,14 @@
 #' @name diagnostics
 NULL
 
+
+#' Check if lintr is 2.0.1.9000 above. A number of features relies on it.
+#' @noRd
+lintr_is_new_enough <- function() {
+    return(compare_version(as.character(utils::packageVersion("lintr")), "2.0.1.9000") >= 0)
```
Comment (renkun-ken) [inline R/diagnostics.R:12]: `utils::packageVersion()` returns a `package_version` object which implements version comparison out of the box. Is there a reason not to directly use `>=` but implements another `compare_version` here?

Comment (randy3k) [reply R/diagnostics.R:12]: Oh, didnt know that. I thought it is only a string. It seems that I have just reinvented the wheel again. Will fix it soon.

**Comment by renkun-ken** [comment]
Also, the lintr version `2.0.1.9000` could be either before or after https://github.com/jimhester/lintr/pull/503 was merged. Therefore, I was using https://github.com/REditorSupport/languageserver/pull/284/files/4d42f4a63d79784340172a484cd6447db6ecbe97#diff-5acd9f0d2237a48b4e62aa0d9ee52a517602b6d4d118ed56187f668f94e19011R93 to detect if `text=` argument is available rather than using a version number.

**Comment by randy3k** [comment]
I have made some changes. Please review again.

File: R/diagnostics.R [hunk 2]
```diff
@@ -5,6 +5,17 @@
 #' @name diagnostics
 NULL
 
+
+#' Check if lintr is 2.0.1.9000 above. A number of features relies on it.
+#' @noRd
+lintr_is_new_enough <- function() {
+    return(
+        utils::packageVersion("lintr") >= "2.0.1.9000" &&
+            "text" %in% methods::formalArgs(lintr::lint)
```
Comment (renkun-ken) [inline R/diagnostics.R:14]: The check failed as we use `methods::` which is not declared in `Import`. We could simpliy use `names(formals(lintr::lint))` without having to import methods as before?

**Review by renkun-ken — APPROVED** [review]
LGTM

**Pull request merged**